streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/TaskMovement.java

```diff
@@ -16,30 +16,35 @@
  */
 package org.apache.kafka.streams.processor.internals.assignment;
 
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.LinkedList;
+import static org.apache.kafka.streams.processor.internals.assignment.HighAvailabilityTaskAssignor.taskIsCaughtUpOnClient;
+
+import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
-import java.util.Set;
 import java.util.SortedSet;
+import java.util.TreeSet;
 import java.util.UUID;
 import org.apache.kafka.streams.processor.TaskId;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class TaskMovement {
-    private static final Logger log = LoggerFactory.getLogger(TaskMovement.class);
+    private static final UUID UNKNOWN = null;
 
     final TaskId task;
-    final UUID source;
-    final UUID destination;
+    private UUID source;
+    private final UUID destination;
 
-    TaskMovement(final TaskId task, final UUID source, final UUID destination) {
+    TaskMovement(final TaskId task, final UUID destination) {
```

Comment:
Do you feel like the source/destination definition has become a bit muddled, or does it still make sense to you? I guess it seems a bit weird to me because in some sense, the source is where we're putting the task now and the destination is where we plan to move it to later. But from another angle, since we now start with the balanced assignment, we are actually moving tasks _from_ the destination _to_ the source.  Does anyone else find that confusing or should I not worry so much

Reply:
We don't need the hashCode/equals implementation anymore, as `TaskMovement` is only ever used in a `SortedSet` with a custom (non-`TaskMovement` based) Comparator. I'll remove it.

That said, we don't really need the `source` field at all anymore. Technically we never used it in the previous code either, but we agreed it made sense to keep for clarity if not usefulness.

But after this refactoring I think the `source` only adds to the confusion, and is never exposed/used for testing. I think at this point it makes sense to remove, but lmk if you disagree.

Reply:
Sure, if we don't need it, then drop it!